src/server/protocol.ts

```diff
@@ -380,27 +397,7 @@ export interface OutliningSpansRequest extends FileRequest {
     command: CommandTypes.GetOutliningSpans;
 }
 
-export interface OutliningSpan {
-    /** The span of the document to actually collapse. */
-    textSpan: TextSpan;
-
-    /** The span of the document to display when the user hovers over the collapsed span. */
-    hintSpan: TextSpan;
-
-    /** The text to display in the editor for the collapsed region. */
-    bannerText: string;
-
-    /**
-     * Whether or not this region should be automatically collapsed when
-     * the 'Collapse to Definitions' command is invoked.
-     */
-    autoCollapse: boolean;
-
-    /**
-     * Classification of the contents of the span
-     */
-    kind: OutliningSpanKind;
-}
+export type OutliningSpan = ChangePropertyTypes<ts.OutliningSpan, { textSpan: TextSpan; hintSpan: TextSpan; }>;

```

Comment:
This is an example of an interface that was copied from services/types.ts with a couple changed property types. I added a utility for this purpose. I only used it in cases where the majority of properties were identical; when there were many changes, I let the near-duplicate stand. Note that this is a homomorphic mapped type so JSDoc from the source type’s properties is retained.